pkg/cmd/pulumi/config.go

```diff
@@ -80,10 +80,188 @@ func newConfigCmd() *cobra.Command {
 	cmd.AddCommand(newConfigRmCmd(&stack))
 	cmd.AddCommand(newConfigSetCmd(&stack))
 	cmd.AddCommand(newConfigRefreshCmd(&stack))
+	cmd.AddCommand(newConfigCopyCmd(&stack))
 
 	return cmd
 }
 
+type stackConfigKeyValues struct {
+	name            string
+	configNamespace string
+	value           string
+	encrypted       bool
+}
+
+func newConfigCopyCmd(stack *string) *cobra.Command {
+	var path bool
+	var newStackName string
+
+	cpCommand := &cobra.Command{
+		Use:   "cp <key>",
+		Short: "Copy config to another stack",
+		Long: "Copies the config from the current stack to the newly specified stack. If a key is omitted,\n" +
+			"then all of the config of the current stack will be copied to the new stack.",
+		Args: cmdutil.MaximumNArgs(1),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			s, err := requireStack(*stack, false, opts, true /*setCurrent*/)
+			if err != nil {
+				return err
+			}
+
+			if s.Ref().Name().String() == newStackName {
+				return errors.New("source and destination stacks must be different when copying config")
+			}
+
+			currentStack, err := loadProjectStack(s)
+			if err != nil {
+				return err
+			}
+
+			configToCopy, err := buildConfigKeysToCopy(args, path, s, currentStack)
+			if err != nil {
+				return err
+			}
+
+			ns, err := requireStack(newStackName, false, opts, false /*setCurrent*/)
+			if err != nil {
+				return err
+			}
+			newStack, err := loadProjectStack(ns)
+			if err != nil {
+				return err
+			}
+
+			var projectNeedsSaved bool
+			for _, key := range configToCopy {
+				err := buildAndSaveConfig(ns, newStack, key)
+				if err != nil {
+					return err
+				}
+				projectNeedsSaved = true
+			}
+
+			if projectNeedsSaved {
+				return saveProjectStack(ns, newStack)
+			}
+
+			return nil
+		}),
+	}
+
+	cpCommand.PersistentFlags().BoolVar(
+		&path, "path", false,
+		"The key contains a path to a property in a map or list to set")
```

Comment:
Does this even work correctly when `--path` is specified? I wonder if we should just not accept `--path` for this command so that only top-level keys can be copied.